src/main/java/frc/robot/arm/Arm.java

```diff
@@ -0,0 +1,114 @@
+package frc.robot.arm;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class Arm extends SubsystemBase {
+    private double targetAngle;
+    private PIDController pid;
+    private boolean broken;
+    private ShuffleboardTab dashboard;
+    private IArmIO armIO;
+
+    public Arm(IArmIO armIO) {
+        this.armIO = armIO;
+        pid = new PIDController(Constants.Arm.PID_CONTROLLER_P_COEFFICIENT, Constants.Arm.PID_CONTROLLER_I_COEFFICIENT, Constants.Arm.PID_CONTROLLER_D_COEFFICIENT);
+        broken = false;
+        initDashboard();
+    }
+
+    public double getAngle() {
+        return armIO.getMeasurement();
+    }
+
+    public boolean isOnTarget() {
+        return ((targetAngle - Constants.Arm.TARGET_MOVEMENT_RANGE_DEGREES <= getAngle()) && (targetAngle + Constants.Arm.TARGET_MOVEMENT_RANGE_DEGREES >= getAngle()));
+    }
+
+    public double getTarget() {
+        return targetAngle;
+    }
+
+    public void setTarget(double inputTarget) {
+        if (inputTarget >= Constants.Arm.MAX_ARM_RANGE_DEGREES) {
+            targetAngle = Constants.Arm.MAX_ARM_RANGE_DEGREES;
+        } else {
+            targetAngle = inputTarget;
+        }
+    }
+
+    public boolean isAtHomePostion() {
+        return armIO.atFrontLimit();
+    }
+
+    public boolean isBroken() {
+        return broken;
+    }
+
+    private String armPositionName() {
+        if (targetAngle == Constants.Arm.HIGH_SCORE_POSITION){
+            return ("High Goal");
+        } else if (targetAngle == Constants.Arm.LOW_SCORE_POSITIOIN){
+            return ("Low Goal");
+        } else if (targetAngle == Constants.Arm.HOME_POSITION) {
+            return ("Home");
+        } else{
+            return ("Moving...");
+        }
+    }
+
+    private boolean armPositionStatus() {
+        if (targetAngle == Constants.Arm.HIGH_SCORE_POSITION || targetAngle == Constants.Arm.LOW_SCORE_POSITIOIN || targetAngle == Constants.Arm.HOME_POSITION){
+            return true;
+        } else{
+            return false;
+        }
+    }
+
+    private void initDashboard() {
+        dashboard = Shuffleboard.getTab("Arm");
+        dashboard.addDouble("Target Angle", () -> getTarget());
+        dashboard.addDouble("Arm Angle", () -> getAngle());
+        dashboard.addBoolean("On Target", () -> armPositionStatus());
+        dashboard.addString("Position", () -> armPositionName());
```

Comment:
These are great things to have on shuffleboard. I think we should also add motor speed as it will could help with debugging (namely the issue we are having right now).